docs/guides/HTTP-Caching.md

```diff
@@ -0,0 +1,69 @@
+
+### HTTP Caching in Tailcall
+
+HTTP caching allows you to cache HTTP responses in Tailcall to improve performance and reduce calls to upstream services.
+
+### What is HTTP Caching?
+HTTP caching refers to storing HTTP responses in a cache so identical requests can be served from the cache instead of making a new API call.
+
+It is a mechanism to reduce latency, network traffic and load on upstream APIs. The cache stores responses keyed by the request URL and headers.
+
+### Overview
+->HTTP caching is disabled by default.
+->To enable, set httpCache to true under upstream.
+->Tailcall uses an LRU cache by default to store responses.
+->The cache respects caching headers like Cache-Control from upstream.
+->Caching reduces calls to upstream APIs for repeated requests.
+
+# Enabling HTTP Caching
+To enable HTTP caching in Tailcall, you need to set httpCache to true under the upstream config:
+
+```upstream:
+  baseURL: "https://api.example.com"
+  httpCache: true
+```
+This will tell Tailcall to cache responses from the upstream API.
+
+### Cache Configuration
+Tailcall uses a Least Recently Used (LRU) cache by default. You can configure the cache size and max age in the Tailcall config file:
+
+To enable HTTP caching globally:
+
+```upstream: 
+  baseURL: https://api.example.com
+  httpCache: true
+```
```

Comment:
Update all examples with with Tailcall DSL

Reply:
@schema
@server(
  port: 4000
)
@upstream(
  baseURL: "https://api.example.com"
  httpCache: true
) {
  query: Query
}.    is this looks good to you @tusharmath .

Reply:
Yes.